service/raft/group0_state_machine.cc

```diff
@@ -83,12 +123,98 @@ future<> group0_state_id_handler::advertise_state_id(utils::UUID state_id) {
     return _gossiper.get().add_local_application_state(gms::application_state::GROUP0_TOMBSTONE_GC_TIME, gms::versioned_value::gc_clock(gc_time));
 }
 
+future<> group0_state_id_handler::on_change(gms::inet_address endpoint, const gms::application_state_map& states, gms::permit_id pid) {
+    return on_application_state_change(
+        endpoint,
+        states,
+        gms::application_state::GROUP0_TOMBSTONE_GC_TIME,
+        pid,
+        [this](gms::inet_address endpoint, const gms::versioned_value& value, gms::permit_id) {
+            const auto gc_time = gc_clock::from_time_t(std::stol(value.value()));
+
+            slogger.debug("Tombstone GC time received: {}, endpoint count: {}", gc_time, _gossiper.get().get_endpoints().size());
+
+            _gc_times[endpoint] = gc_time;
+
+            auto* const group0_server = _server_accessor.get_server();
+            if (!group0_server) {
+                slogger.debug("Group0 server not found");
+                return make_ready_future<>();
+            }
+
+            const auto& group0_members = group0_server->get_configuration().current;
```

Comment:
We could skip reacting to the notification if we are in the middle of a joint group 0 configuration (i.e. a configuration change, i.e. `previous` is non-empty)

Because if we are in a joint config, there might be some servers that are currently leaving. (Present in `previous` but not in `current`). In that case, I am not sure if it's 100% safe to do the calculation only based on `current` members.

OTOH if we skip, we might miss the notification completely (and arbitrary time could pass before another notification arrives). For correctness we'd have to queue it up and then fire when the group 0 config change happens.

Perhaps the safest solution would be to take the sum of `previous` and `current`.

Reply:
But the more I think of it... perhaps this handler should be reacting to group 0 configuration changes too?

Consider: a server is leaving. All but the leaving servers apply a command that updates state_id. We get notifications, but we fail to update `_gc_time_last_reconcile` because of that server. Eventually, the server leaves, however we still don't update `_gc_time_last_reconcile` -- we don't react to the config change.

Generally, if we calculate some value based on multiple sources of data, then reacting to only one of the sources changing is not completely correct.

Reply:
done - changed to react to the group0 configuration changes.